src/radical/pilot/tmgr/staging_input/default.py

```diff
@@ -239,10 +240,13 @@ def work(self, tasks):
 
                     tmp_path = tempfile.mkdtemp(prefix='rp_agent_tar_dir')
                     tmp_dir  = os.path.abspath(tmp_path)
-                    tar_name = '%s.%s.tar' % (self._session.uid, self.uid)
-                    tar_tgt  = '%s/%s'     % (tmp_dir, tar_name)
+                    tar_name = '%s.%s.%04d.tar' % (self._session.uid, self.uid,
```

Comment:
wouldn't it be a too long name? (`sid + tar_id` seems enough, and `self.uid` is of the component, and not of the task)

Reply:
Yes, the uid is not strictly needed, but helps to distinguish from the tarballs created by the launcher base (which also contains `self._uid`).